Gettext_helpers.cmake

```
@@ -97,8 +97,8 @@ function(configure_gettext)
                 "--output=${GETTEXT_POTFILE_DESTINATION}/${GETTEXT_DOMAIN}.pot"
             WORKING_DIRECTORY "${CMAKE_CURRENT_SOURCE_DIR}")
     endif()
-    add_custom_command(
-        OUTPUT "${GETTEXT_POTFILE_DESTINATION}/${GETTEXT_DOMAIN}.pot"
+    # should use add_custom_target so .pot doesn't get cleaned
+    add_custom_target(${TARGET_NAME}_gen_pot
         COMMAND "${GETTEXT_XGETTEXT_COMMAND}" ${GETTEXT_XGETTEXT_ARGS}
             ${GETTEXT_SOURCES}
             "--output=${GETTEXT_POTFILE_DESTINATION}/${GETTEXT_DOMAIN}.pot"
```

```suggestion
            ${GETTEXT_SOURCES}
            "--output=${GETTEXT_POTFILE_DESTINATION}/${GETTEXT_DOMAIN}.pot"
            "--join-existing"
            "--force-po"
```
`join-existing` improves upon the idea of this PR by not throwing out previously created `.pot`, while `force-po` makes the generation more consistent.

While these flags could be passed through `XGETTEXT_ARGS` they are not really project-dependent like, say, `sort-output`. Additionally it would require anyone using this project to comb through gettext util's documentation.